### PR TITLE
improved mount commands

### PR DESCRIPTION
**Pull request by mapk-amazon**
### Description of changes
* Fixed the Access point for EFS commands

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb [hunk 1]
```diff
@@ -43,14 +43,18 @@
...
-    elsif efs_encryption_in_transit == "true"
+    if efs_encryption_in_transit == "true"
       mount_options += ",tls"
+      # iam authorization requires tls
       if efs_iam_authorization == "true"
         mount_options += ",iam"
       end
+      # accesspoint requires tls
+      if efs_access_point_id
+        mount_options += ",accesspoint=#{efs_access_point_id},"
```
Comment (dreambeyondorange) [inline cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb]: following the pattern, this probably shouldn't have a comma at the end

File: cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb [hunk 2]
```diff
@@ -37,7 +37,7 @@
     # get efs to unmount
     node.default['cluster']['unmount_efs_shared_dir_array'], node.default['cluster']['unmount_efs_fs_id_array'], = get_efs(UNMOUNT_ACTION)
     # get efs to mount
-    node.default['cluster']['mount_efs_shared_dir_array'], node.default['cluster']['mount_efs_fs_id_array'], node.default['cluster']['mount_efs_encryption_in_transit_array'], node.default['cluster']['mount_efs_iam_authorization_array'] = get_efs(MOUNT_ACTION)
+    node.default['cluster']['mount_efs_shared_dir_array'], node.default['cluster']['mount_efs_fs_id_array'], node.default['cluster']['mount_efs_encryption_in_transit_array'], node.default['cluster']['mount_efs_iam_authorization_array'], node.default['cluster']['mount_efs_access_point_array'] = get_efs(MOUNT_ACTION)
```
Comment (dreambeyondorange) [inline cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb:40]: Were you able to test this update logic?

File: cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb (hunk 1 above)
Comment (mapk-amazon) [reply cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb]: You are absolutely right. Was smuggled in by some internal merging on my side 👍

File: cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb (hunk 2 above)
Comment (mapk-amazon) [reply cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb:40]: For me it only mounted EFS successfully after providing this code change. Not sure whether "real" updating will work (This is why i set it to unsupported for now).

Comment (dreambeyondorange) [reply cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb:40]: If its set to unsupported, this logic should probably be removed, unless it's required for updates regardless of efs access points?

**Review by dreambeyondorange — APPROVED** [review]

**Pull request merged**